test/test_torch.py

```diff
@@ -12539,6 +12539,28 @@ def test_topk_noncontiguous_gpu(self, device):
         self.assertEqual(top1, top2)
         self.assertEqual(idx1, idx2)
 
+    @dtypes(torch.int8, torch.uint8, torch.int16, torch.int32, torch.int64)
+    def test_topk_integral(self, device, dtype):
+        a = torch.randint(torch.iinfo(dtype).min, torch.iinfo(dtype).max, size=(10,),
+                          dtype=dtype, device=device)
+        sort_topk = a.sort()[0][-5:].flip(0)
+        topk = a.topk(5)
+        self.assertEqual(sort_topk, topk[0])      # check values
+        self.assertEqual(sort_topk, a[topk[1]])   # check indices
+
+    def test_topk_nonfinite(self, device):
+        for dtype in (torch.half, torch.float, torch.double):
```

Comment:
Hi can you use `dtypes()` decorator instead of the for loops over dtypes here? 
Decorator is generally more preferred way for this, as downstream backends can optionally skip all tests with one specific dtype (like XLA for torch.half). 
Thanks!

Reply:
Lol Natalia [commented the same thing](https://github.com/pytorch/pytorch/pull/35734#discussion_r400601607) on the PR against master.  I'll push that change to both, didn't get to it immediately because I'm juggling a lot of unrelated communication stuff right now (getting the word out about upstream Amp).

Reply:
Changed to
```
    @dtypesIfCUDA(torch.half, torch.float, torch.double)
    @dtypes(torch.float, torch.double)
    def test_topk_nonfinite(self, device, dtype):
```
I don't expect this will cause bfloat16 to be tested on XLA.  @ailzhang if you want bfloat16 tested on XLA, is there an alternative decorator I should use?